static/preview-code.js

```diff
@@ -0,0 +1,8 @@
+import hljs from 'https://cdn.jsdelivr.net/npm/highlightjs@9.16.2/+esm'
+
+const inscription = document.documentElement.dataset.inscription;
+const response = await fetch(`/content/${inscription}`);
+const code = await response.text();
+const codeBlock = document.createElement('code');
+codeBlock.innerHTML = hljs.highlightAuto(code).value;
```

Comment:
```suggestion
codeBlock.innerHTML = hljs.highlight(code).value;
```
Maybe use only `highlight` and pass in the content type explicitly. You can get the content type from the response object I think

Reply:
We will have to remember to update the supported languages if we add anymore in `media.rs`. See update